src/test/java/uk/gov/hmcts/dts/fact/services/CourtServiceTest.java

```diff
@@ -368,6 +378,34 @@ void shouldReturnListForNearestCourtsByPostcodeSearch() {
         assertThat(results.getCourts().get(0)).isInstanceOf(CourtReferenceWithDistance.class);
     }
 
+    @Test
+    void shouldReturnEmptyListIfNoMapitdataForPostcodeOnlySearch() {
+        when(mapitService.getMapitData(any())).thenReturn(empty());
+        List<CourtReferenceWithDistance> results = courtService.getNearestCourtReferencesByPostcode(
+            JE2_4BA
+        );
+        assertEquals(0, results.size());
+        verifyNoInteractions(serviceAreaSearchFactory);
+    }
+
+    @Test
+    void shouldReturnListIfMapitdataForPostcodeOnlySearchFound() {
+        final MapitData mapitData = mock(MapitData.class);
+        when(mapitService.getMapitData(JE2_4BA)).thenReturn(Optional.of(mapitData));
+
+        final List<uk.gov.hmcts.dts.fact.entity.CourtWithDistance> courts = asList(
+            mock(uk.gov.hmcts.dts.fact.entity.CourtWithDistance.class),
+            mock(uk.gov.hmcts.dts.fact.entity.CourtWithDistance.class));
+        when(proximitySearch.searchWith(mapitData)).thenReturn(courts);
+
+        List<CourtReferenceWithDistance> results = courtService.getNearestCourtReferencesByPostcode(
+            JE2_4BA
+        );
+        assertEquals(2, results.size());
+        assertThat(results.get(0)).isInstanceOf(CourtReferenceWithDistance.class);
+        assertThat(results.get(1)).isInstanceOf(CourtReferenceWithDistance.class);
```

Comment:
Don't think we need the second assertion here. This is a list of objects. If the first object is of a particular class, the second object will be the same too